ui/packages/plot/src/Plot.svelte

```diff
@@ -1,20 +1,55 @@
-<script>
+<script lang='ts'>
 	//@ts-nocheck
 	import Plotly from "plotly.js-dist-min";
 	import { Plot as PlotIcon } from "@gradio/icons";
+	import { colors as color_palette, ordered_colors } from "@gradio/theme";
+	import { get_next_color } from "@gradio/utils";
 	import { Vega } from "svelte-vega";
-
 	import { afterUpdate, onDestroy } from "svelte";
+	import {create_config} from "./utils"
 
 	export let value;
 	export let target;
 	let spec = null;
+	export let colors: Array<string> = [];
+	export let theme: string;
+	export let caption: string;
+	
+	function get_color(index: number) {
```

Comment:
I yanked this from `Chart.svelte`. Ideally this would be consolidated but it relies on the global `colors` prop so I wasn't sure what the best way to refactor was.

Reply:
This function should produce the same colors for qualitative scales as the `Chart.svelte` (e.g. `TimeSeries`) component which is why I used it here

Reply:
We'll probably want to update the color schemes for visualisations in the future with a few pre-baked options for users. Data-viz colour schemes have some specific requirements/ demands that fall outside of typical 'design' issues. We can address that (and the duplication) later.

Reply:
Completely agree! Filed #2780